pkg/cmd/root/root.go

```diff
@@ -63,21 +83,55 @@ func NewRootCmd(cfg *config.Config) *cobra.Command {
 		"Log level - writes to ./corectl.log if set",
 	)
 
-	appCmd, err := application.NewAppCmd(cfg)
+	rootCmd.PersistentFlags().BoolVar(
+		&nonInteractive,
+		"nonint",
+		false,
+		"Run in non-interactive mode - the command will error if it needs to ask for user input",
+	)
+	rootCmd.PersistentFlags().BoolVar(
+		&nonInteractive,
+		"non-interactive",
+		false,
+		"Run in non-interactive mode - the command will error if it needs to ask for user input",
+	)
+
+	// --non-interactive is the standard used by other clis
+	err := rootCmd.PersistentFlags().MarkDeprecated("nonint", "please use --non-interactive instead.")
 	if err != nil {
-		panic("Unable to execute app command")
+		log.Panic().Msg("unable to set --nonint as deprecated")
 	}
-	p2pCmd, err := p2p.NewP2PCmd(cfg)
-	if err != nil {
-		panic("Unable to execute p2p command")
+
+	if cfg.IsPersisted() {
```

Comment:
Do we really want to do it?
I think it should not be necessary to run the initialization on CI, for example. 
It should be possible to simply specify everything corectl needs via flags

Reply:
it currently *is* necessary to run init on CI, these commands already exit if the config isn't set

Reply:
Can you give an example?
It shouldn't be the case, unless you didn't set up any of the necessary flags

Reply:
```
➜  corectl git:(feature/update) ✗ ./corectl tenant create --dry-run
...
Error: cplatform repository path is not set. consider initializing corectl first:
  corectl config init
```

```
➜  corectl git:(feature/update) ✗ ./corectl app create test --tenant michael --dry-run
INFO: resetting local repository [repo=]
INFO: resetting local repository [repo=]
📝 Creating new application test: https://github.com//test
                                         
  1. <empty>                             
  2. blank                               
> 3. go-web                              
...

Error: couldn't load tenant configuration in path tenants/tenants: stat .: no such file or directory
```

```
➜  corectl git:(feature/update) ✗ ./corectl p2p export --tenant michael -e gcp-dev
Usage:
  corectl p2p export [flags]
...
Error: cplatform repository path is not set. consider initializing corectl first:
  corectl config init
```

I can see now that you *could* set these via the params (I don't know if this is very likely outside of scripts), I'll roll this back